### PR TITLE
doc/develop: Add picolibc documentation

### LINKED ISSUE #49620: Add picolibc documentation
**Is your enhancement proposal related to a problem? Please describe.**

Picolibc integration support has been added in #44096, but it is missing documentation.

**Describe the solution you'd like**

Add "Picolibc" page under [Developing with Zephyr/Language Support/C Language Support/Standard Library](https://docs.zephyrproject.org/latest/develop/languages/c/index.html#standard-library).

### PR DESCRIPTION
**Pull request by keith-packard**
This describes the capabilities offered by Picolibc along with some of the requirements applications have when using it.

Closes: #49620

Signed-off-by: Keith Packard <keithp@keithp.com>

### CODE REVIEW

File: doc/develop/languages/c/picolibc.rst [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+Picolibc
+########
+
+`Picolibc`_ is a complete C library implementation written for the
+embedded systems, targeting `C17 (ISO/IEC 9899:2018)`_ and `POSIX 2018
+(IEEE Std 1003.1-2017)`_ standards. Picolibc is an external open
+source project which is provided for Zephyr as a module, and included
+as part of the :ref:`toolchain_zephyr_sdk` in precompiled form for
+each supported architecture (:file:`libc.a`)
```
Comment (mbolivar-nordic) [inline doc/develop/languages/c/picolibc.rst:11]: Missing period at end of sentence

File: doc/develop/languages/c/picolibc.rst [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+Picolibc Module
+===============
+
+When built as a Zephyr module, there are several configuration knobs
+available to adjust the feature set in the library, balancing what the
+library supports versus the code size of the resulting
+functions. Because the standard C++ library must be compiled for the
+target C library, the module version cannot be used with applications
+which need that library. Building the module version will increase the
```
Comment (mbolivar-nordic) [inline doc/develop/languages/c/picolibc.rst:34]: I'm having trouble understanding the reference in `that library` -- do you mean the C++ or the C library? I think C++, right?

**Review by mbolivar-nordic — COMMENTED** [review]
Nice work! I found one typo and have one question

File: doc/develop/languages/c/picolibc.rst (hunk 2 above)
Comment (keith-packard) [reply doc/develop/languages/c/picolibc.rst]: Yeah, sorry for the ambiguous reference. I've made it explicit.

**Review by mbolivar-nordic — APPROVED** [review]

**Review by stephanosio — APPROVED** [review]
Great work, thanks!

**Comment by mbolivar-nordic** [comment]
Merging on @carlescufi 's behalf

**Pull request merged**